### PR TITLE
Fix type hints

### PR DESCRIPTION
**Pull request by jl-wynen**

### CODE REVIEW

File: src/scitacean/dataset.py [hunk 1]
```diff
@@ -396,7 +398,7 @@ def _get_or_add_orig_datablock(self, key: Union[int, str]) -> OrigDatablock:
 
     def make_upload_model(self) -> Union[UploadDerivedDataset, UploadRawDataset]:
```
Comment (YooSunYoung) [inline src/scitacean/dataset.py:399]: Can `UploadDerivedDataset` and `UploadRawDataset` be `TypeVar` instead?
If the model is `Type[UploadRawDataset]`, the returned model is `UploadRawDataset` for example, right...?

Comment (jl-wynen) [reply src/scitacean/dataset.py:399]: But this method doesn't take any arguments except `self`. So how would you match the type of the model?

Comment (YooSunYoung) [reply src/scitacean/dataset.py:399]: Oh Sorry I thought the model was also an input. Never mind...

File: src/scitacean/filesystem.py [hunk 2]
```diff
@@ -86,7 +88,7 @@ def to_local(self) -> PurePath:
 
     def __truediv__(self, other: Union[str, RemotePath]) -> RemotePath:
         """Join two path segments."""
-        if isinstance(other, (PurePath, Path)):
+        if isinstance(other, (PurePath, Path)):  # type: ignore[unreachable]
             raise TypeError("OS paths are not supported when concatenating RemotePath.")
```
Comment (YooSunYoung) [inline src/scitacean/filesystem.py:92]: I have a question how a `type-checker` would decide unreachable...
will this line below will also be considered unreachable...?
```
if not isinstance(other, (str, RemotePath)) and isinstance(other, (PurePath, Path)):
    raise ...
```

**Review by YooSunYoung — APPROVED** [review]

File: src/scitacean/filesystem.py (hunk 2 above)
Comment (jl-wynen) [reply src/scitacean/filesystem.py:92]: I guess so. It must be based on the argument type hints. And mypy then assumes that no invalid type is ever passed in.

**Pull request merged**